### PR TITLE
Update Spanner sample to show close of resources.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 1]
```diff
@@ -273,32 +273,36 @@ static void deleteExampleData(DatabaseClient dbClient) {
 
   // [START spanner_query_data]
   static void query(DatabaseClient dbClient) {
-    // singleUse() can be used to execute a single read or query against Cloud Spanner.
-    ResultSet resultSet =
+    // We use a try-with-resource block to automatically release resources held by ResultSet.
+    try (ResultSet resultSet =
         dbClient
```
Comment (lesv) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Is it really easiest to read with this on a seperate line from the = ?

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 2]
```diff
@@ -273,32 +273,36 @@ static void deleteExampleData(DatabaseClient dbClient) {
...
+            // singleUse() can be used to execute a single read or query against Cloud Spanner.
```
Comment (lesv) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Strike the word and put it on the same line as the method.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 3]
```diff
@@ -273,32 +273,36 @@ static void deleteExampleData(DatabaseClient dbClient) {
...
   // [START spanner_read_data]
   static void read(DatabaseClient dbClient) {
-    ResultSet resultSet =
+    // We use a try-with-resource block to automatically release resources held by ResultSet.
+    try (ResultSet resultSet =
         dbClient
             .singleUse()
             .read(
                 "Albums",
                 // KeySet.all() can be used to read all rows in a table. KeySet exposes other
```
Comment (lesv) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Again, can this be shortened to be on the same line as KeySet.All() ?

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 4]
```diff
@@ -466,30 +472,33 @@ static void queryUsingIndex(DatabaseClient dbClient) {
...
+            resultSet.isNull("MarketingBudget") ? "NULL" : resultSet.getLong("MarketingBudget"));
+      }
     }
   }
   // [END spanner_query_data_with_index]
 
   // [START spanner_read_data_with_index]
   static void readUsingIndex(DatabaseClient dbClient) {
-    ResultSet resultSet =
+    // We use a try-with-resource block to automatically release resources held by ResultSet.
```
Comment (lesv) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: try-with-resources isn't that novel these days.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 5]
```diff
@@ -646,61 +662,67 @@ static void updateWithTimestamp(DatabaseClient dbClient) {
   // [START spanner_query_data_with_timestamp_column]
   static void queryMarketingBudgetWithTimestamp(DatabaseClient dbClient) {
     // Rows without an explicit value for MarketingBudget will have a MarketingBudget equal to
-    // null.
-    ResultSet resultSet =
+    // null. A try-with-resource block is used to automatically release resources held by
+    // ResultSet.
+    try (ResultSet resultSet =
```
Comment (lesv) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: would it be easier to read if dbClient was on this line?

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 1 above)
Comment (jsimonweb) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Updated.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 2 above)
Comment (jsimonweb) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Done.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 3 above)
Comment (jsimonweb) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Done.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 4 above)
Comment (jsimonweb) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Opting to leave these comments in for now.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 5 above)
Comment (jsimonweb) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Updated.

**Pull request merged**